build.sc

```diff
@@ -94,14 +94,30 @@ object main extends MillModule {
 
     def ivyDeps = Agg(
       // Keep synchronized with ammonite in Versions.scala
-      ivy"com.lihaoyi:::ammonite:1.6.7",
+      ivy"com.lihaoyi:::ammonite:1.6.9",
       // Necessary so we can share the JNA classes throughout the build process
       ivy"net.java.dev.jna:jna:4.5.0",
       ivy"net.java.dev.jna:jna-platform:4.5.0"
     )
 
     def generatedSources = T {
-      Seq(PathRef(shared.generateCoreSources(T.ctx().dest)))
+      val dest = T.ctx().dest
+      val version = publishVersion()
+      writeBuildInfo(dest, version)
+      shared.generateCoreSources(dest)
+      Seq(PathRef(dest))
+    }
+
+    def writeBuildInfo(dir : os.Path, version : String) = {
```

Comment:
I don't disagree with the principle, but BuildInfo being a contrib module, I don't think building mill should depend on it. 

Moreover, my personal opinion is that the `BuildInfo` sbt plugin only exists because people don't understand sbt enough to write that logic themselves. Mill having much simpler semantics, everybody can write the logic and I don't think there's a justification to have a "BuildInfo" plugin. 

That being said, you're a maintainer and I'm not, so if you're convinced this PR should make use of the buildInfo module, I'll abide (ironically though, I'll have to hardcode the version as this PR hasn't been merged yet )

Reply:
Good point. Yet, we repeat ourselves when we do it "manually". So, please treat my request as not mandatory.